corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js

```diff
@@ -1259,12 +1312,12 @@ hqDefine("cloudcare/js/form_entry/entries", function () {
             case constants.BINARY:
                 if (!toggles.toggleEnabled('WEB_APPS_UPLOAD_QUESTIONS')) {
                     // do nothing, fall through to unsupported
+                } else if (style === constants.SIGNATURE) {
+                    entry = new SignatureEntry(question, {});
+                    break;
                 } else {
                     switch (question.control()) {
                         case constants.CONTROL_IMAGE_CHOOSE:
-                            if (question.stylesContains(constants.SIGNATURE)) {
-                                break;
```

Comment:
Should the conditional check and creation `entry = new SignatureEntry(question, {});` be kept inside the switch for consistency?

But I do think we need to keep `question.stylesContains(constants.SIGNATURE)` instead of just `style === constants.SIGNATURE`. The latter seems like it'll if an appearance attribute is configured for the question (such as applying a '2-per-row' appearance attribute for question tile.

Reply:
Good catch on using `stylesContains`, you're definitely right.

I don't think it necessarily needs to be in the `switch(question.control())` though, since it's not based on having the control number for "image". It makes sense to me to not put it inside another conditional if it doesn't need to be. But I guess then the question is, is there any reason to be concerned with using a "signature" appearance attribute on non-"image" file upload questions? My gut says there's no reason that would ever happen, but that if it does, it's probably fine to let it turn it into a signature capture, right?

Reply:
> using a "signature" appearance attribute on non-"image" file upload questions? 
What would this error look like - does it still allow you to submit the form but the signature doesn't get saved?

>  since it's not based on having the control number for "image"
I'm interpreting the signature as just a special type of image, so we expect it to always have an image control number right? My preference is to limit the scope if possible but not blocking btw

Reply:
> > using a "signature" appearance attribute on non-"image" file upload questions?
> 
> What would this error look like - does it still allow you to submit the form but the signature doesn't get saved?

As an attribute under `constants.BINARY` but not limited by `CONTROL_IMAGE_CHOOSE`, it seems to just work without error - it would generate a `SignatureEntry` and get saved properly as a file upload by Formplayer.

> I'm interpreting the signature as just a special type of image, so we expect it to always have an image control number right? My preference is to limit the scope if possible but not blocking btw

With this reasoning and considering how weird it might be to try to debug (for example) an "Audio" question that's getting displayed as a signature entry, I think you're right that it's better to limit scope here. https://github.com/dimagi/commcare-hq/pull/33481/commits/e672c5a5a31cd31e59fb55cdf29d11d6b231060b

Reply:
Cool, thanks for checking